packages/cli/src/build/config.ts

```diff
@@ -172,22 +191,136 @@ export function getRollupConfigs(pkg: Package, aliases: Aliases) {
       });
     });
 
+  let hasBrowserCondition = false;
+  let hasWorkerCondition = false;
+  if (pkg.json.preconstruct.exports) {
+    if (typeof pkg.entrypoints[0].json.exports == "object") {
```

Comment:
This sort of looking at `pkg.entrypoints[0].json.exports` for the exports field isn't right, the root might not be an entrypoint, it should be `pkg.json.exports`

Reply:
Thanks for the feedback. I'm still getting familiar with this code base. I'll work on cleaning this up.

Reply:
Ok, I'm looking at this again and I'm not totally sure I understand the change you would like me to make -- please forgive the ignorance. The reason this is authored this way is because I was trying to keep with the conventions that I saw earlier in this file. If you look at the statements above this one, they use `pkg.entrypoints[0].json[someField]` to check for certain conditions.

Am I missing something where the way I'm doing it doesn't apply in this case? Can you help me better understand the refactoring that you would like to see?

Reply:
I think that I understand what Mitchell means and I've wanted to write a failing test case for the existing code on the main branch. However, I've realized that I don't quite know how such a test would look like.

So maybe there is no actual bug with the current implementation - although it's a little bit quirky how it works. I will explain that in a sec.

With `package.json#exports` we have a clear single place where all entrypoints can be declared, both for our config and for external consumption. So we should use `pkg.json.exports` here to check for this because this is that clear place to declare this - so conceptually this makes sense and is the most correct thing to do, semantically-wise. On top of that, there is a chance that somebody uses such a config:
```json
{
  "preconstruct": {
    "entrypoints": [
      "foo.js",
      "bar.js"
    ]
  }
}
```

and thus the `pkg.entrypoints[0].json.exports` won't hold the information that we need. It's even possible that somebody does this:
```json
{
  "preconstruct": {
    "entrypoints": [
      "foo.js",
      "index.js"
    ]
  }
}
```
Notice that here we still have a root entrypoint, but it's not the first one.

So why there is no problem with the existing code and why we don't refer to `pkg.json.exports` there? Well, cause if somebody doesn't include the root entrypoint then the root package.json won't have those fields like browser, module, and stuff. It might not even have a main field! If allowing for `require('pkg')` is not desirable (for whatever reason) then it doesn't make sense to include any of those fields in its root `package.json`... cause event if we did, where would they point to?

So the current logic merely ensures that all existing entrypoints have the same build types declared on them and the order in which they appear in the `entrypoints` array doesn't really matter. There is no clear single place for those fields to be declared so we kinda "fallback" to comparing against the first entrypoint - but that can be just any entrypoint.

This raises an interesting problem - those other validations should possibly be extended with the knowledge about the `pkg.json.exports` and they also should check that (only if the opt-in flag is enabled, of course). This should validate that if for example there is `pkg.entrypoints[i].browser` then there also should be `pkg.json.exports[entryPointPath].browser` (and vice-versa).

Reply:
Ok, I *think* I understand this better and I went ahead and made the change. I now look for the exports fields on the package and not on the entrypoint.